activerecord/lib/active_record/connection_adapters/abstract/connection_handler.rb

```diff
@@ -227,6 +262,26 @@ def set_pool_manager(connection_name)
           connection_name_to_pool_manager[connection_name] ||= PoolManager.new
         end
 
+        def pool_managers
+          connection_name_to_pool_manager.values
+        end
+
+        def deprecation_for_pool_handling(method)
+          roles = []
+          pool_managers.each do |pool_manager|
+            roles << pool_manager.role_names
+          end
+
+          if roles.flatten.uniq.count > 1
+            ActiveSupport::Deprecation.warn(<<-MSG.squish)
+              The `#{method}` method is being called on all connection pools,
+              regardless of role. If you intended to only clear connections on a
+              specific role, ensure that role is passed in. To turn off this deprecation
+              warning pass, `:all`.
```

Comment:
IMO we want the deprecation warning path to continue acting as it does today, until they accept the new behavior with `:all` (or opt out with an explicit role)

```suggestion
              `#{method}` currently only applies to connection pools in the current
              role (`#{ActiveRecord::Base.current_role}`). In Rails 7.1, this method
              will apply to all known pools, regardless of role. To affect only those
              connections belonging to a specific role, pass the role name as an
              argument. To switch to the new behavior, pass `:all` as the role name.
```

(or something along those lines -- more a suggestion of intent than specific wording)

Reply:
Done